### ISSUE TITLE
[Quickstart suggestion]: Government - Multilingual Government Content Summarization & Generation Assistant

### ISSUE THREAD

**Issue by mhdawson** [rh-ai-kickstart/ai-kickstart-contrib#35]
### Quickstart suggestion

# Multilingual Government Content Summarization & Generation Assistant

Government departments process vast amounts of content daily—parliamentary debates, policy reports, stakeholder consultations, and departmental publications—that must be distilled into actionable briefings for decision-makers. This quickstart demonstrates an automated content summarization and distribution pipeline built on OpenShift AI that monitors incoming documents, generates bilingual summaries with quality controls, and distributes targeted briefings to appropriate recipients. Based on real Canadian government use cases like ParlBrief (parliamentary committee summaries) and Senior Management Briefs (media monitoring), this solution addresses the second-largest AI use case in the Government of Canada AI Register: content generation and summarization (76 projects, 18.6%).

This quickstart showcases enterprise-grade AI capabilities not covered in existing demonstrations: comprehensive guardrails (factual consistency checking, PII detection, bias monitoring), true multilingual workflows (parallel EN/FR processing with translation quality assurance), template-based content generation (government-specific formats like Treasury Board briefing notes), and production drift detection (monitoring quality metrics over time). Using real Canadian parliamentary data (Hansard) and open government publications, developers will build a complete pipeline from document ingestion through summarization, translation, validation, and distribution, demonstrating how OpenShift AI handles government-grade summarization with compliance, bilingual support, and quality controls that go beyond existing RAG/chatbot examples.

```mermaid
graph LR
    A[Incoming Content<br/>News, Reports, Meetings] --> B[Summarize<br/>Extract Key Points]
    B --> C[Translate<br/>EN ↔ FR]
    C --> D[Distribute<br/>To Recipients]

    style A fill:#e1f5ff
    style B fill:#fff4e1
    style C fill:#ffe1f5
    style D fill:#e1ffe1
```

## Why This Use Case

**Second-Largest Demand**: Content Generation & Summarization represents 76 projects (18.6%) in the [Government of Canada AI Register (Open Data Portal)](https://open.canada.ca/data/en/dataset/fcbc0200-79ba-4fa4-94a6-00e32facea6b) - the second most common use case after search.

**Critical Government Applications**:
- Summarizing lengthy policy documents and reports
- Drafting briefing notes for executives
- Newsletter and communication generation
- Synthesizing consultation responses
- Executive summary creation

**Unique Government Requirement**: Multilingual (English/French, or other language combinations) support is non-negotiable for federal government communications, making this a distinct value-add that generic summarization tools don't address well.

### What Exists in RH AI Quickstarts:
- RAG chatbots (search & retrieval)
- Fraud detection
- Code generation
- Basic LLM serving
- Agentic automation of IT processes

 ## What Exists in RH AI Quickstarts:
- RAG chatbots (search & retrieval)
- Fraud detection
- Code generation
- Basic LLM serving
- Agentic automation of IT processes

### What's Missing:
- **No dedicated summarization pipeline** with quality controls
- **No bilingual/translation-aware workflows**
- **No government-specific content generation** with compliance guardrails
- **No executive summary generation** with templating

### Real GC AI Register Examples (Content Generation & Summarization - 6 Projects):

These projects from the [Government of Canada AI Register (Open Data Portal)](https://open.canada.ca/data/en/dataset/fcbc0200-79ba-4fa4-94a6-00e32facea6b) demonstrate the need for the full workflow.

1. **ParlBrief** (ISED) - Automates transcription and summarization of bilingual parliamentary committee meetings. Delivers summaries to Privy Council Office within 3 hours of meeting end. Uses language models to transcribe audio, translate content, generate summaries, and assess tone. Addresses high overtime costs and staff burnout from manual summarization.

2. **Senior Management Briefs** (GAC) - Automates monitoring and summarization of media coverage. Scans and interprets a wide range of news sources, extracts key developments, and produces concise actionable summaries for senior management distribution.

3. **d@Media** (ESDC) - Drafts internal media monitoring newsletters using AI to filter and summarize relevant news articles based on predefined requirements. Automates newsletter generation for employee distribution.

4. **Missions AI-Generated Briefs** (GAC) - Streamlines monthly synthesis of Trade-related events and activities reported by Canadian missions across the United States. Transforms reports into concise, regionally structured summaries for staff, enabling more time acting on insights rather than preparing them.

5. **DataMinr First Alert - ReGenAI Feature** (PCO) - Collects publicly available information, selects what's relevant, bundles into user-friendly summaries with translation if needed. Provides real-time dynamically generated event briefs, summarizing and contextualizing events as they happen for efficient consumption of event monitoring responsibilities.

6. **Executive Correspondence Workflow Automation Tool** (ISED) - Semi-automates generation of bilingual synopses of ministerial emails (5,500+ annually). Uses large language models to generate summaries in both English and French with consistent format and tone. Supports categorization and routing to appropriate sector. Reduces manual workload and improves timeliness of executive correspondence processing.

### Canadian Hansard (Parliamentary Debates)

The ideal dataset for this quickstart - provides authentic bilingual Canadian government content with natural summarization tasks.

**Why Hansard is Perfect**:
- ✅ **Truly bilingual**: Parallel English/French translations by professional government translators
- ✅ **Government domain**: Actual Canadian parliamentary proceedings
- ✅ **Natural summarization task**: Long debates naturally require summarization (précis)
- ✅ **Large scale**: Decades of continuously growing data
- ✅ **Publicly available**: Open government data with permissive licensing
- ✅ **Well-structured**: XML/CSV formats with speaker attribution, topics, metadata
- ✅ **Real use case**: Library of Parliament's Parliamentary Précis project does exactly this

**Available Sources**:

1. **Lipad Project (1901-2019)**
   - URL: https://www.lipad.ca/data/
   - Comprehensive searchable Hansard dataset
   - Formats: PostgreSQL dumps, UTF-8 CSV, XML
   - Coverage: Complete historical record from 1901-2019
   - License: Open access for research and development

2. **House of Commons Official Open Data (2006-Present)**
   - URL: https://www.ourcommons.ca/en/open-data
   - Current and historical debates in XML format
   - Coverage: 39th Parliament onwards (April 3, 2006 - present)
   - Both official languages published simultaneously
   - Updated daily with new proceedings
   - License: Open Government License - Canada

3. **University of Alberta Dataverse (2006-2015)**
   - URL: https://dataverse.library.ualberta.ca/dataverse/hansard
   - Research-ready curated dataset
   - Well-documented and cleaned for NLP tasks
   - License: Freely available for academic/research use

**Suggested Summarization Tasks Using Hansard**:

1. **Committee Hearing Summaries**
   - Input: 3-hour committee transcript (50-100 pages)
   - Output: Executive summary with witness testimonies, key questions, recommendations
   - Matches real Library of Parliament Parliamentary Précis use case

2. **Daily Proceedings Digest**
   - Input: Full day's House debates (100-200 pages)
   - Output: Brief with key motions, votes, ministerial statements
   
3. **Topic-Focused Synthesis**
   - Input: Week's worth of debates on specific topic (e.g., healthcare, budget)
   - Output: Thematic summary with positions, arguments, decisions

4. **Bilingual Newsletter**
   - Input: Notable debates and decisions from past week
   - Output: Citizen-friendly newsletter in both official languages

5. **Speaker-Specific Summaries**
   - Input: All speeches by a specific MP or Minister
   - Output: Position summary and key arguments
## Proposed Architecture

```
┌─────────────────────────────────────────────────────────┐
│  Input Layer                                             │
│  - Single document upload (PDF, DOCX, TXT)               │
│  - Multi-document batch processing                       │
│  - URL ingestion (web articles, reports)                 │
│  - Language detection (EN/FR/Mixed)                      │
└─────────────────────────────────────────────────────────┘
                           ↓
┌─────────────────────────────────────────────────────────┐
│  Document Preprocessing                                  │
│  - Text extraction and chunking                          │
│  - Language segmentation (handle mixed EN/FR docs)       │
│  - Structure detection (headings, sections, tables)      │
│  - Metadata extraction (date, author, department)        │
└─────────────────────────────────────────────────────────┘
                           ↓
┌─────────────────────────────────────────────────────────┐
│  Summarization Engine                                    │
│  - Abstractive summarization                             │
│  - Extractive summarization (key sentence selection)     │
│  - Configurable length (executive/detailed/bullet)       │
│  - Section-aware summarization (preserve structure)      │
└─────────────────────────────────────────────────────────┘
                           ↓
┌─────────────────────────────────────────────────────────┐
│  Bilingual Processing Layer                              │
│  - Translation (EN↔FR) with domain preservation          │
│  - Terminology consistency (government lexicons)         │
│  - Parallel summary generation (both languages)          │
│  - Quality assurance (translation validation)            │
└─────────────────────────────────────────────────────────┘
                           ↓
┌─────────────────────────────────────────────────────────┐
│  Content Generation & Templating                         │
│  - Briefing note templates                               │
│  - Newsletter formatting                                 │
│  - Executive summary templates                           │
│  - Custom template engine (department-specific)          │
│  - Citation preservation and linking                     │
└─────────────────────────────────────────────────────────┘
                           ↓
─────────────────────────────────────────────────────────┐
│  Guardrails & Compliance Layer                          │
│  - Factual consistency checking (no hallucinations)      │
│  - PII detection and masking                             │
│  - Sentiment analysis (ensure neutral tone)              │
│  - Citation verification                                 │
│  - Bias detection                                        │
│  - Drift detection                                       │
│  - Human-in-the-loop review workflow                     │
└─────────────────────────────────────────────────────────┘
                           ↓
┌─────────────────────────────────────────────────────────┐
│  Output & Export Layer                                  │
│  - DOCX/PDF export with formatting                       │
│  - Side-by-side EN/FR display                            │
│  - Version control and audit trail                       │
│  - Quality metrics dashboard                             │
└─────────────────────────────────────────────────────────┘
```
### 1. Bilingual Excellence
- **Parallel Processing**: Generate summaries in both EN and FR simultaneously
- **Government Terminology**: Use official translation memories and lexicons
- **Mixed Document Handling**: Process documents with both languages
- **Quality Validation**: Ensure translations maintain original intent

### 2. Summarization Modes
- **Executive Summary** (1-2 paragraphs, high-level takeaways)
- **Detailed Summary** (sections with key points, configurable length)
- **Bullet Point Summary** (actionable items, decisions, recommendations)
- **Section-Specific** (summarize only specific sections)

### 3. Content Generation Templates
- **Briefing Notes** - Standard TB/PCO format with Issue/Background/Analysis/Recommendation
- **Newsletter Articles** - Engaging summaries for internal/external communications
- **Executive Dashboards** - Key metrics and insights from reports
- **Consultation Summaries** - Synthesize stakeholder feedback themes
- **Committee Evidence** - Parliamentary summary format

### 4. Quality & Compliance
- **Factual Consistency Scoring**: Compare summary against source using NLI models
- **Citation Tracking**: Link summary statements to source paragraphs
- **Hallucination Detection**: Flag generated content not supported by source
- **PII Detection**: Automatically detect and mask sensitive information
- **Bias Detection**: Ensure neutral, objective tone for government use
- **Confidence Scoring**: Provide quality metrics for human review

### 5. Enterprise Integration
- **Batch Processing**: Process multiple documents in pipeline
- **API-First Design**: REST API for integration with GCDocs, SharePoint
- **Audit Logging**: Full traceability of all summarization operations
- **User Feedback Loop**: Collect corrections to improve models
- **Role-Based Access**: Department-specific templates and controls

### 1. Parliamentary Committee Summary
**Input**: 3-hour committee hearing transcript from Hansard (50-100 pages)
**Output**: Structured summary with witness testimonies, MP questions, key themes in EN and FR
**Features**: Speaker attribution, topic clustering, controversy detection, bilingual output
**Data Source**: Canadian Hansard - Committee proceedings (1901-present)
**Example**: Standing Committee on Health hearing on pandemic preparedness

### 2. Departmental Report Executive Brief
**Input**: 100-page departmental annual report from Open Government Portal
**Output**: 2-page executive brief in EN and FR with key findings, recommendations, impacts
**Features**: Auto-extract performance metrics, risk analysis, financial impacts, bilingual template
**Data Source**: Canada Open Government Portal - Departmental reports and publications
**Example**: Environment and Climate Change Canada Annual Report → Executive summary

### 3. Multi-Topic Parliamentary Digest
**Input**: Week's worth of Hansard debates on specific topic (e.g., budget, healthcare) - 200+ pages
**Output**: Thematic analysis with party positions, key arguments, areas of consensus/conflict in EN and FR
**Features**: Party-based sentiment analysis, theme clustering, representative quote extraction, debate timeline
**Data Source**: Canadian Hansard - Daily debates filtered by topic
**Example**: All Budget 2024 debates from February-March → Consolidated position summary

### 4. Daily Parliamentary Proceedings Brief
**Input**: Full day's House of Commons debates from Hansard (100-200 pages)
**Output**: Executive digest with key motions, votes, ministerial statements in both official languages
**Features**: Priority detection, motion tracking, template-based formatting, link to full debates
**Data Source**: Canadian Hansard - Daily proceedings (updated daily)
**Example**: February 4, 2026 proceedings → One-page digest for distribution to stakeholders

## OpenShift AI / Llama Stack Feature Mapping

This section maps Proposal 2's requirements to OpenShift AI and Llama Stack features, highlighting which capabilities are **NEW** (not yet covered by existing quickstarts) versus **REUSED** (already demonstrated elsewhere).

### Features This Quickstart Will Use/Demonstrate

| Category | Feature | Status in Existing Quickstarts | How Proposal 2 Uses It | New or Reused |
|----------|---------|-------------------------------|------------------------|---------------|
| **Model Serving (Inference)** | **KServe (Single-model Serving)** | ✅ Published<br>Used by: Product-recommender, RAG, and others | Serve open-source LLM for summarization and NLLB-200/mBART for translation | **REUSED** |
| **Operations & Orchestration** | **Data Science Pipelines** | ✅ Published<br>Used by: Product-recommender, RAG | Orchestrate document processing workflow:<br>- Document ingestion<br>- Summarization<br>- Translation (EN↔FR)<br>- Quality validation<br>- Template generation<br>- Distribution | **REUSED** |
| **Governance & Trust** | **Guardrails Orchestrator** | ❌ Not published as standalone<br>Some quickstarts use: guardrailing-llms, lemonade-stand-assistant | **NEW USE CASE**: Integrate guardrails for:<br>- Factual consistency checking (NLI model)<br>- PII detection (Presidio)<br>- Bias detection<br>- Tone enforcement (neutral, professional)<br>- Citation verification | **✨ NEW - Extended Usage** |
| **Governance & Trust** | **Bias Detection** | ❌ Not covered | **NEW**: Monitor and flag bias in generated summaries to ensure neutral government communications | **✨ NEW** |
| **Governance & Trust** | **Drift Detection** | ❌ Not covered | **NEW**: Monitor quality metrics (factual consistency, correction rates, bilingual performance) over time. Detect statistically significant drift using KS test. Alert when metrics breach thresholds. Dashboard shows trends. **Detection only - no automated remediation.** | **✨ NEW** |
| **Connectivity & Ecosystem** | **Llama Stack Operator** | ⚠️ Y/N - Multiple quickstarts use Llama Stack but not OpenShift AI Operator | Could use for unified deployment of Llama models | **REUSED** (if used) |
| **Accelerator Support** | **vLLM / Intel Gaudi** | ✅ Published<br>Used by: RAG, product-recommender, and others | Accelerate inference for summarization and translation models | **REUSED** |

### Features NOT used/demonstrated

These OpenShift AI/Llama Stack features are not demonstrated in the quickstart:

| Feature | Existing Quickstart Coverage | Why Not Demonstrated in Proposal 2 |
|---------|------------------------------|-------------------------------|
| **Training Operator** | ❌ None | Using pre-trained models (open-source LLMs, NLLB-200); no model training in this quickstart |
| **Workbenches (Jupyter Notebooks)** | ⚠️ Partial (not as OpenShift AI workbench)<br>- billing-extraction-with-GroundX<br>- basic-speech-to-text-with-whisper<br>- Byo-agentic-framework<br>- Data-governance-co-pilot | Using pre-trained models directly; no experimentation or fine-tuning needed |
| **Pre-built Images** | ✅ Yes<br>- Billing-extraction-with-GroundX (uses OpenShift AI workbench image: Jupyter, Minimal, CPU, Python 3.12) | Not doing model development; deploying existing models only |
| **InstructLab Integration** | ❌ None | Not creating new models; using existing open models |
| **ModelMesh (Multi-model Serving)** | ❌ None (being deprecated) | Being deprecated; KServe sufficient for our needs |
| **Models-as-a-Service (MaaS)** | ❌ None | Using local deployment for government data sovereignty; MaaS not suitable for sensitive government data |
| **LLM Compressor** | ❌ None | Model size (8B parameters) manageable without compression |
| **LLM-D (Semantic Router)** | ✅ Yes<br>- LLM Semantic Router | Not needed; straightforward summarization workflow without intelligent routing between multiple models |
| **Distributed Workloads (Ray & CodeFlare)** | ⚠️ Used by Llama Stack distributions<br>(Not a standalone feature - embedded in distributions) | Single-node inference sufficient; not doing distributed training or large-scale batch processing |
| **Model Registry (GA)** | ❌ None | Nice-to-have but not critical for MVP; can add in Phase 4 |
| **Model Catalog** | ❌ None | Using well-known open-source models; catalog not needed |
| **Model Context Protocol (MCP)** | ✅ Yes<br>**Llama Stack feature - Published**<br>- It-self-service-agent<br>- Ai-virtual-agent<br>- ai-architecture-charts | Not integrating with external context providers; using local data sources only |
| **Explainability** | ❌ None | Citation linking provides transparency; formal explainability not required |
| **Structured Output** (Llama Stack) | ❌ None | Not constraining output to specific JSON schemas; using template-based formatting instead |
| **Advanced Sampling** (Llama Stack) | ❌ None | Default sampling parameters sufficient for summarization |
| **Multimodal Support** (Llama Stack) | ✅ Yes<br>- ai-virtual-agent | Text-only summarization; not processing images or multi-modal content |
| **Shield API** (Llama Stack) | ✅ Yes<br>- It-self-service-agent<br>- ai-virtual-agent | Using Guardrails Orchestrator instead for safety/compliance |
| **Telemetry** (Llama Stack) | ✅ Yes<br>- lls-observability<br>- It-self-service-agent | Using Prometheus/Grafana for monitoring instead |
| **Evaluation API** (Llama Stack) | ❌ None | Using custom quality metrics (factual consistency, correction rates) instead |

**Comment by mhdawson** [comment]
We likely need to trim down the total features/what is covered in the quickstart but this starts with a fuller description of what a solution might include
